src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File fileFromFileName = new File(fromFileName);
+        File fileToFileName = new File(toFileName);
+        Map<String, Integer> result = new HashMap<>();
+
+        if (!fileFromFileName.exists()) {
+            return;
+        }
+
+        readFromFileAndPutDataToMap(fileFromFileName.getName(), result);
+
+        writeToFileDataFromMap(fileToFileName.getName(), result);
+    }
+
+    private void writeToFileDataFromMap(String toFileName, Map<String, Integer> result) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            int supply = result.getOrDefault("supply", 0);
+            int buy = result.getOrDefault("buy", 0);
+            int finalResult = supply - buy;
+
+            bufferedWriter.write("supply," + supply);
+            bufferedWriter.newLine();
+            bufferedWriter.write("buy," + buy);
+            bufferedWriter.newLine();
+            bufferedWriter.write("result," + finalResult);
+        } catch (IOException ioException) {
+            throw new RuntimeException("Can't write data to the file " + toFileName, ioException);
+        }
+    }
+
+    private void readFromFileAndPutDataToMap(String fromFileName, Map<String, Integer> result) {
```

Comment:
You should split this method into two separate. One for reading from the file, Second to create report (Or, as in your case, to change the read data from the file and write it to the map). Here is an example how read method should look:

![image](https://github.com/user-attachments/assets/a303acb9-8bba-4d51-958e-ba20089e47ed)

Reply:
![image](https://github.com/user-attachments/assets/b6ba51a4-2cf1-40f1-9b04-897962177f5d)